### PR TITLE
APIGOV-18728 Use the agent-sdk Job package to poll for mulesoft analytics events

### PR DESCRIPTION
**Pull request by tsjohns9**
Created a new struct that implements the Job interface and wraps the MuleEventEmitter to hook the traceability polling into the sdk.

### CODE REVIEW

File: pkg/traceability/muleemitter.go [hunk 1]
```diff
@@ -2,89 +2,150 @@ package traceability
...
+	consecutiveErrors int
+	cfg               *config.AgentConfig
+	done              chan bool
+	eventChannel      chan string
+	pollInterval      time.Duration
+	jobID             string
+}
+
+// MuleEventEmitterJob wraps an Emitter and implements the Job interface so that it can be executed by the sdk.
+type MuleEventEmitterJob struct {
```
Comment (tsjohns9) [inline pkg/traceability/muleemitter.go:38]: Added a new struct that wraps the MuleEventEmitter. It implements the Emitter and the Job interface.

File: pkg/traceability/muleemitter.go [hunk 2]
```diff
@@ -2,89 +2,150 @@ package traceability
...
+	}
+	return false
+}
+
+func traceabilityHealthCheck(name string) *hc.Status {
+	status := &hc.Status{
+		Result: hc.OK,
+	}
+	// Check the status of the connection to mulesoft
+	health := hc.GetStatus(anypoint.HealthCheckEndpoint)
```
Comment (tsjohns9) [inline pkg/traceability/muleemitter.go:145]: The health check is using the same check performed by the client to see if the connection to mulesoft is alive. Should we also check if the connection to condor is good too?

I'm also a little unsure if this is the right way to implement this check. The mulesoft client registers a health check for this endpoint, `anypoint.HealthCheckEndpoint`. And this endpoint just gets the result from that. Is that fine @jcollins-axway  @vivekschauhan

File: default_mulesoft_traceability_agent.yml [hunk 3]
```diff
@@ -1,34 +1,71 @@
...
+      publicKey: "${CENTRAL_AUTH_PUBLICKEY:/keys/public_key.pem}"
+      realm: "${CENTRAL_AUTH_REALM:Broker}"
+      url: "${CENTRAL_AUTH_URL:https://login.axway.com/auth}"
+    ssl:
+      cipherSuites: ${CENTRAL_SSL_CIPHERSUITES:["ECDHE-ECDSA-AES-256-GCM-SHA384", "ECDHE-RSA-AES-256-GCM-SHA384", "ECDHE-ECDSA-CHACHA20-POLY1305", "ECDHE-RSA-CHACHA20-POLY1305", "ECDHE-ECDSA-AES-128-GCM-SHA256", "ECDHE-RSA-AES-128-GCM-SHA256", "ECDHE-ECDSA-AES-128-CBC-SHA256", "ECDHE-RSA-AES-128-CBC-SHA256"]}
+      insecureSkipVerify: ${CENTRAL_SSL_INSECURESKIPVERIFY:false}
+      maxVersion: ${CENTRAL_SSL_MAXVERSION:"TLS1.3"}
+      minVersion: ${CENTRAL_SSL_MINVERSION:"TLS1.2"}
+      nextProtos: ${CENTRAL_SSL_NEXTPROTOS:[]}
+    gateway-section:
```
Comment (jcollins-axway) [inline default_mulesoft_traceability_agent.yml:30]: This gateway-section is meant for the mulesoft settings

File: pkg/traceability/muleemitter.go (hunk 2 above)
Comment (jcollins-axway) [reply pkg/traceability/muleemitter.go:145]: This is fine, as the sdk should regularly be validating that the connection is up.  The job is just piggy backing on what the health check library is doing.

**Pull request merged**